### PR TITLE
[1] Use rule definition in issue conversion

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Part of #2189

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/CFamily/CFamilyIssueToAnalysisIssueConverterTests.cs [hunk 1]
```diff
@@ -369,9 +369,6 @@ public void ConvertFromIssueSeverity_InvalidValue_Throws()
         public void ConvertFromIssueType(IssueType cfamilyIssueType, AnalysisIssueType analysisIssueType)
         {
             CFamilyIssueToAnalysisIssueConverter.Convert(cfamilyIssueType).Should().Be(analysisIssueType);
-
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/CFamily/CFamilyIssueToAnalysisIssueConverterTests.cs:372]: Misc: noticed this when blatantly copying the tests; it's a redundant assertion that belongs to the test below.

File: src/TypeScript.UnitTests/Analyzer/EslintBridgeIssueConverterTests.cs [hunk 2]
```diff
@@ -18,18 +18,22 @@
...
 using SonarLint.VisualStudio.TypeScript.Analyzer;
 using SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract;
+using SonarLint.VisualStudio.TypeScript.Rules;
 
 namespace SonarLint.VisualStudio.TypeScript.UnitTests.Analyzer
 {
     [TestClass]
     public class EslintBridgeIssueConverterTests
     {
+        private const string TestEslintBridgeRuleId = "rule id";
```
Comment (rita-sonarsource) [inline src/TypeScript.UnitTests/Analyzer/EslintBridgeIssueConverterTests.cs:35]: A bit ugly but I don't have a neat way to mock the collection and it's simpler.

File: src/TypeScript/Analyzer/EslintBridgeIssueConverter.cs [hunk 3]
```diff
@@ -32,28 +33,29 @@ internal interface IEslintBridgeIssueConverter
...
     internal delegate string ConvertToSonarRuleKey(string eslineRuleKey);
+    internal delegate IEnumerable<RuleDefinition> GetRuleDefinitions();
 
     internal class EslintBridgeIssueConverter : IEslintBridgeIssueConverter
     {
         private readonly ConvertToSonarRuleKey keyMapper;
+        private readonly GetRuleDefinitions ruleDefinitionProvider;
 
-        public EslintBridgeIssueConverter(ConvertToSonarRuleKey keyMapper)
+        public EslintBridgeIssueConverter(ConvertToSonarRuleKey keyMapper, GetRuleDefinitions ruleDefinitionProvider)
```
Comment (duncanp-sonar) [inline src/TypeScript/Analyzer/EslintBridgeIssueConverter.cs:43]: Strictly speaking this should be using the current active rule configuration (assuming we allow the user to change the severity; they can for C#, and I think they can for C++).
However, we can change that when we support user configuration.

File: src/TypeScript.UnitTests/Analyzer/EslintBridgeIssueConverterTests.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/TypeScript.UnitTests/Analyzer/EslintBridgeIssueConverterTests.cs:35]: I think it's ok, but I'd change the name to better indicate that the value isn't important as it isn't checked; it just needs to be valid e.g. `ValidEsLintKey`, `AnyEslintKey`.

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**